constraint-solver/src/solver/constraint_splitter.rs

```diff
@@ -0,0 +1,565 @@
+use std::{
+    fmt::Display,
+    ops::{Add, Div},
+};
+
+use itertools::Itertools;
+use num_traits::Zero;
+use powdr_number::{FieldElement, LargeInt};
+
+use crate::{
+    constraint_system::AlgebraicConstraint,
+    grouped_expression::{GroupedExpression, RangeConstraintProvider},
+    runtime_constant::RuntimeConstant,
+};
+
+/// Tries to split the given algebraic constraint into a list of equivalent
+/// algebraic constraints.
+/// This is the case for example if the variables in this expression can
+/// be split into different bit areas.
+/// The general strategy is to find an offset and a multiplier such that
+/// the equation is equivalent to an equation in the integers,
+/// and then apply a modulo operation to find independent parts.
+pub fn try_split_constraint<T: RuntimeConstant + Display, V: Clone + Ord + Display>(
+    constraint: &AlgebraicConstraint<&GroupedExpression<T, V>>,
+    range_constraints: &impl RangeConstraintProvider<T::FieldType, V>,
+) -> Option<Vec<AlgebraicConstraint<GroupedExpression<T, V>>>> {
+    let (quadratic, linear, constant) = constraint.expression.components();
+    if !quadratic.is_empty() {
+        // We cannot split quadratic constraints.
+        return None;
+    }
+    let mut constant = constant.try_to_number()?;
+
+    // Group the linear part by absolute coefficients.
```

Comment:
It would help to have an explanation of what absolute coefficients are and what grouping is. Maybe a small example in the comment?